File: src/main/java/com/pluralsight/rxjava2/nitrite/dataaccess/ProductDataAccess.java
package com.pluralsight.rxjava2.nitrite.dataaccess;

import com.pluralsight.rxjava2.nitrite.entity.Product;
import io.reactivex.rxjava3.core.Observable;
import org.dizitart.no2.Nitrite;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class ProductDataAccess {

    private final static Logger log = LoggerFactory.getLogger(ProductDataAccess.class);

    public static Observable<Product> select(Nitrite db) {
        return Observable.create(emitter -> {

            try {

                for (Product nextProduct : db.getRepository(Product.class)
                        .find()) {
                    //log.info(nextProduct.toString());
                    emitter.onNext(nextProduct);
                }
                emitter.onComplete();
            }
            catch( Throwable t ) {
                emitter.onError(t);
            }
        });
    }
}

File: src/main/java/com/pluralsight/rxjava2/nitrite/dataaccess/ProductDataAccess.kt
package com.pluralsight.rxjava2.nitrite.dataaccess

import com.pluralsight.rxjava2.module2.log
import com.pluralsight.rxjava2.nitrite.entity.Product
import io.reactivex.rxjava3.core.Observable
import io.reactivex.rxjava3.core.ObservableEmitter
import org.dizitart.no2.Nitrite

object ProductDataAccess {
    fun select(db: Nitrite): Observable<Product?> {
        return Observable.create { emitter: ObservableEmitter<Product?> ->
            try {
                for (nextProduct in db.getRepository(Product::class.java)
                        .find()) {
                    log.info(nextProduct.toString());
                    emitter.onNext(nextProduct)
                }
                emitter.onComplete()
            } catch (t: Throwable) {
                emitter.onError(t)
            }
        }
    }
}